Documentation/trouble-index.rst

```diff
@@ -21,17 +22,20 @@ contains target block offset and generation that last changed this block. The
 block it points to then upon read verifies that the block address and the
 generation matches. This check is done on all tree levels.
 
-The number in **faled on 30736384** is the logical block number, **wanted 10**
+The number in **failed on 30736384** is the logical block number, **wanted 10**
 is the expected generation number in the parent node, **found 8** is the one
 found in the target block.  The number difference between the generation can
 give a hint when the problem could have happened, in terms of transaction
 commits.
 
-Once the mismatched generations are stored on the device, it's permanent and
-cannot be easily recovered, because of information loss. The recovery tool
-``btrfs restore`` is able to ignore the errors and attempt to restore the data
-but due to the inconsistency in the metadata the data need to be verified by the
-user.
+Once the mismatched generations are stored on the device, without a good copy
+from another replica, it's permanent and cannot be easily recovered because of
+information loss. However, if a valid copy exists on another replica, btrfs will
+transparently correct the read error, and running ``btrfs scrub`` in read-write
+mode will fix the error permanently by copying the valid metadata block over the
+invalid one. Otherwise, the recovery tool ``btrfs restore`` is able to ignore
```

Comment:
Nowadays we have `-o rescue=all,ro` mount option, which is a more useful solution compared to `btrfs restore`

Thus we may want to promote the new solution, and use `btrfs restore` as the last option (and under most cases it's not that better compared to rescue=all)